File: src/backend/orchestrator/src/service/profile_client.py
import os
import grpc

import proto.query.profile.query_profile_pb2 as query_profile_pb2
import proto.query.profile.query_profile_pb2_grpc as query_profile_pb2_grpc

from google.protobuf import field_mask_pb2

profile_address=os.environ['profile-address']
profile_port=os.environ['profile-port']

channel_string = f"{profile_address}:{profile_port}"

class ProfileClient:

  def __ProfileData(datas):
    print(f'__ProfileData datas:{datas}', flush=True)
    return [ query_profile_pb2.ProfileData( **data ) for data in datas if data ]
  
  def __init__(self):
    # instantiate a channel
    self.channel  = grpc.insecure_channel(channel_string)
    # bind the client and the server
    self.stub = query_profile_pb2_grpc.ProfileStub(self.channel)

  def get_profile_by_id(self, id, paths):
    datas=[{'id': id}]
    return self.get_profile( datas=datas, paths=paths )

  def get_profile(self, datas, paths):
    #print(f'get_profile 1', flush=True)
    profiles = ProfileClient.__ProfileData(datas=datas)

    #print(f'get_profile 2', flush=True)
    response = self.stub.Get(
              query_profile_pb2.ProfilesRequest(
                profilesData=query_profile_pb2.ProfileDataList(
                    profiles=profiles
                  ),
                fields=field_mask_pb2.FieldMask( 
                  paths=paths 
                )
              )
            )
    #print(f'get_profile 3', flush=True)
    return response

File: src/frontend/frontend_service/src/grpc/query/ProfileClient.js
import * as grpcWeb from 'grpc-web';
import { FieldMask } from 'google-protobuf/google/protobuf/field_mask_pb';
import { Empty } from 'google-protobuf/google/protobuf/empty_pb';

import {
  ProfileClient, 

  ProfileData,
  ProfileDataList,
  ProfilesRequest
} from '../../proto/query_webclient_grpc_web_pb';

String.prototype.capitalize = function() {
  return this.charAt(0).toUpperCase() + this.slice(1);
}

const ProfileService = new ProfileClient(
                          'https://' + window.location.hostname + ':443',
                          null, null);              

const IdentificationProfile = ( { result, error=null, metadata={} } ) => {

  const request = new Empty();

  const call = ProfileService.identification(
    request, metadata, 
    (err, response) => {
      if (err) {
        console.log(`Unexpected error for IdentificationProfile: code = ${err.code}, message = "${err.message}"`);
        if (error){
          error(err.message)
        }
        result()
      } else {
        console.log("IdentificationProfile not error... Get response");
        console.log("IdentificationProfile", response.toObject() );
        result(
          response.toObject()
        )
      }
  });
}

//rpc Get (ProfilesRequest) returns (ProfileDataList);
const GetProfile = ( {data, result, error=null, paths=[], metadata={}} ) => {

  const profile = new ProfileData()
  Object.entries(data).forEach( ([key, value]) => {
    profile["set"+key.capitalize()](value)
  })

  const fields = new FieldMask();
  fields.setPathsList( paths );

  const profiles = new ProfileDataList()
  profiles.setProfilesList(
    [ profile ]
  )

  const request = new ProfilesRequest();
  request.setProfilesdata( profiles );
  request.setFields( fields );

  const call = ProfileService.get(
    request, metadata, 
    (err, response) => {
      if (err) {
        console.log(`Unexpected error for GetProfile: code = ${err.code}, message = "${err.message}"`);
        if (error){
          error({
            msg: err.message,
            type: 'ErrorMessage'
          })
        }
        result()
      } else {
        console.log("GetProfile", response.toObject().profilesList );
        if (error){
          error()
        }
        result(
          response.toObject().profilesList
        )
      }
  });
}

export { 
  GetProfile,
  IdentificationProfile,
};